hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1713,6 +1713,12 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	}
 	if capiProviderDeploymentSpec != nil {
 		proxy.SetEnvVars(&capiProviderDeploymentSpec.Template.Spec.Containers[0].Env)
+		for i := range capiProviderDeploymentSpec.Template.Spec.Containers {
```

Comment:
all the changes in this file will trigger a rollout for those components across the fleet which might result in a resource consumption spike. We probably want to call this out for managed services consumers